.github/workflows/main.yml

```diff
@@ -1,40 +1,75 @@
-# This is a basic workflow to help you get started with Actions
+---
+name: Run Tests
 
-name: CI
-
-# Controls when the action will run.
 on:
-  # Triggers the workflow on push or pull request events but only for the main branch
   push:
     branches: [ main, 'v*']
   pull_request:
     branches: [ main, 'v*' ]
+  release:
+    types: [released]
 
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
-
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+      - uses: pre-commit/action@v3.0.0
+        with:
+          extra_args: -a
+
   tests:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        django-version: ["3.2.0", "4.1.3"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
 
-      - name: Set up Python 3.9
-        uses: actions/setup-python@v1
+      #----------------------------------------------
+      #        load pip cache if cache exists
+      #----------------------------------------------
+      - uses: actions/cache@v3
         with:
-          python-version: 3.9
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip
+          restore-keys: ${{ runner.os }}-pip
 
-      - name: Run image
-        uses: abatilo/actions-poetry@v2.0.0
+      #----------------------------------------------
+      #  -----  install & configure poetry  -----
+      #----------------------------------------------
+      - name: Install Poetry
+        uses: abatilo/actions-poetry@v2
+      #----------------------------------------------
+      #       load cached venv if cache exists
+      #----------------------------------------------
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
```

Comment:
Had another review cause of my PR
I found out that this step will never cache due to path not existing
![image](https://user-images.githubusercontent.com/5886584/210121761-0ba0d7a7-6d75-4915-a6cc-b3c48d6e3768.png)

And therefore it will never load
![image](https://user-images.githubusercontent.com/5886584/210121767-fb44cd99-e883-4875-8536-3ed2f96d910b.png)

Reply:
Thanks! TBO I just copied it from another project of mine... :upside_down_face: